# this patch injects firefox-gnome-theme in autoconfig
# this is how we getting gnome theme loaded after browser started
# also it replaces /etc/mobile-config-firefox with /app/etc/mobile-config-firefox
# we heed it because flatpak uses /app/etc instead of /etc
--- a/./src/mobile-config-autoconfig.js
+++ b/./src/mobile-config-autoconfig.js
@@ -146,7 +146,7 @@ function css_file_get_fragments(name) {
         return g_fragments_cache[name];

     var ret = [];
-    var path = "/etc/mobile-config-firefox/" + name + ".files";
+    var path = "/app/etc/mobile-config-firefox/" + name + ".files";
     log("Reading fragments from file: " + path);
     var file = new FileUtils.File(path);

@@ -160,7 +160,7 @@ function css_file_get_fragments(name) {
         var line = {};
         has_more = istream.readLine(line);
         if (css_fragment_check_firefox_version(line.value))
-            ret.push("/etc/mobile-config-firefox/" + line.value);
+            ret.push("/app/etc/mobile-config-firefox/" + line.value);

     } while (has_more);

@@ -186,7 +186,7 @@ function css_file_get(name) {
 // file: return of css_file_get()
 function css_file_delete_outdated(name, file) {
     var depends = css_file_get_fragments(name).slice(); /* copy the array */
-    depends.push("/etc/mobile-config-firefox/" + name + ".files");
+    depends.push("/app/etc/mobile-config-firefox/" + name + ".files");
     for (var i in depends) {
         var depend = depends[i];
         var file_depend = new FileUtils.File(depend);
@@ -214,6 +214,8 @@ function css_file_merge(name, file) {
                   createInstance(Components.interfaces.nsIFileOutputStream);
     ostream.init(file, 0x02 | 0x08 | 0x20, 0644, 0);

+    write_line(ostream, '@import url("file:///app/lib/firefox-gnome-theme/' + name + '.css");');
+
     var fragments = css_file_get_fragments(name);
     for (var i in fragments) {
         var line;

